### ISSUE TITLE
Deal with new default branch name

### ISSUE THREAD

**Issue by iamed2** [invenia/PkgTemplates.jl#233]
https://github.com/github/renaming

New repos on GitHub will have their default branch set to "main" instead of "master" starting October 1, 2020. 

I think the best thing to do would be to make this change now, as having an alternate default branch name is already well-supported.

**Comment by mschauer** [comment]
The copy paste-instructions for making a new repo on GitHub
```
git remote add origin https://github.com/mschauer/MitosisStochasticDiffEq.jl.git
git branch -M main
git push -u origin main
```
now change the default branch underhandedly, which creates problems with the remote already added. :-/

**Comment by simeonschaub** [comment]
The Documenter template will also need to respect the specified branch, right now Documenter defaults to only deploying docs when on the branch `master`.

**Comment by christopher-dG** [comment]
Let's try to figure out what actually needs doing:

```
$ rg master src templates
src/plugins/ci.jl
136:    "https://travis-ci.com/{{{USER}}}/{{{PKG}}}.jl.svg?branch=master",
310:        "https://{{{HOST}}}/{{{USER}}}/{{{PKG}}}.jl/badges/master/pipeline.svg",
315:        "https://{{{HOST}}}/{{{USER}}}/{{{PKG}}}.jl/badges/master/coverage.svg",
316:        "https://{{{HOST}}}/{{{USER}}}/{{{PKG}}}.jl/commits/master",

src/plugins/license.jl
9:  [here](https://github.com/invenia/PkgTemplates.jl/tree/master/templates/licenses).

src/plugins/coverage.jl
21:    "https://codecov.io/gh/{{{USER}}}/{{{PKG}}}.jl/branch/master/graph/badge.svg",
43:    "https://coveralls.io/repos/github/{{{USER}}}/{{{PKG}}}.jl/badge.svg?branch=master",
44:    "https://coveralls.io/github/{{{USER}}}/{{{PKG}}}.jl?branch=master",

templates/appveyor.yml
22:    - master

templates/cirrus.yml
12:    - sh -c "$(fetch https://raw.githubusercontent.com/ararslan/CirrusCI.jl/master/bin/install.sh -o -)"

templates/gitlab-ci.yml
50:    - master
```

- The occurrences in `license.jl` and `cirrus.yml` don't need to be changed. 
- The badge URLs in the CI and coverage plugins will need to depend on the default branch name, as will the entries in the AppVeyor and GitLab CI configs.
- The Git plugin itself already respects Git's `init.defaultBranch` setting, but it doesn't actually store the value anywhere so other plugins can't know what it is. Maybe it can be made to store the default branch as a field, and then other plugins can access the value through it. That does mean we have to choose a "default default" branch, for cases where `init.defaultBranch` isn't set. Git's default is still `master`, but we should probably use `main`.

> The Documenter template will also need to respect the specified branch, right now Documenter defaults to only deploying docs when on the branch master.

Excellent point, we'll set `devbranch` explicitly in `make.jl`.

**Comment by nickrobinson251** [comment]
is there anything left to do here post-https://github.com/invenia/PkgTemplates.jl/pull/316 ?

**Comment by nickrobinson251** [comment]
closing as fixed (#316)
